src/server/handlers/join.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: MIT
+//! Create subnet handler and parameters
+
+use crate::manager::SubnetManager;
+use crate::server::handlers::subnet::SubnetManagerPool;
+use crate::server::JsonRPCRequestHandler;
+use anyhow::anyhow;
+use async_trait::async_trait;
+use fvm_shared::address::Address;
+use fvm_shared::econ::TokenAmount;
+use ipc_sdk::subnet_id::SubnetID;
+use ipc_subnet_actor::JoinParams;
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+use std::sync::Arc;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct JoinSubnetParams {
+    pub subnet: String,
+    pub from: Option<String>,
+    pub collateral: u64,
+    pub validator_net_addr: String,
+}
+
+/// The create subnet json rpc method handler.
+pub(crate) struct JoinSubnetHandler {
+    pool: Arc<SubnetManagerPool>,
+}
+
+impl JoinSubnetHandler {
+    pub(crate) fn new(pool: Arc<SubnetManagerPool>) -> Self {
+        Self { pool }
+    }
+}
+
+#[async_trait]
+impl JsonRPCRequestHandler for JoinSubnetHandler {
+    type Request = JoinSubnetParams;
+    type Response = ();
+
+    async fn handle(&self, request: Self::Request) -> anyhow::Result<Self::Response> {
+        let conn = match self.pool.get(&request.subnet) {
```

Comment:
This is a bug, you are getting the connection to the subnet ID, not to the parent, and you want to interact with the subnet actor in the parent.

Reply:
updated!